docs/source/extensions.rst

```diff
@@ -3,17 +3,233 @@
+        'n_failed': gym.spaces.Box(0, np.inf, (1,), np.int32),
+        'n_wifi': gym.spaces.Box(1, np.inf, (1,), np.int32),
+        'power': gym.spaces.Box(-np.inf, np.inf, (1,)),
+        'cw': gym.spaces.Discrete(32767),
+        'mcs': gym.spaces.Discrete(12)
+    })
+
+Next, we define the **parameter function** that will provide the initial power value for agents that require
+this parameter as a constructor argument. We can do this by creating an appropriate method and decorating it with
+the ``@parameter`` decorator. Parameter function are methods of the extension class and cannot take any additional
+arguments:
+
+.. code-block:: python
+
+    @parameter
+    def initial_power(self):
+        return 16.0206
+
+We can also specify type of the returned value in the OpenAI Gym format. It will help the library to check if
+a given value type is compatible with the argument type required by the agent:
+
+.. code-block:: python
+
+    @parameter(parameter_type=gym.spaces.Box(-np.inf, np.inf, (1,)))
+    def initial_power(self) -> float:
+        return 16.0206
+
+Note that name of the function must match name of the argument required by the agent. If there already exists
+a function with that name, we can name the function differently and explicitly define the argument name in
+the decorator:
+
+.. code-block:: python
+
+    @parameter(parameter_name='initial_power', parameter_type=gym.spaces.Box(-np.inf, np.inf, (1,)))
+    def default_pow(self) -> float:
+        return 16.0206
+
+We define the **observation functions** analogous to parameter functions. The only differences are that we use
+the ``@observation`` decorator and that the implemented method takes additional parameters. Below is an
+example observation function that provides approximated collision probability in dense IEEE 802.11ax networks:
+
+.. code-block:: python
+
+    @observation
+    def pc(self, n_wifi, *args, **kwargs):
+        return 0.154887 * np.log(1.03102 * n_wifi)
+
+Note that the observation function can take parameters that are specified in the observation space.
+:ref:`BaseExt <base_ext>` methods will automatically pass the given observation to the function to allow
+dynamic computation of the returned value. What is important, observation methods take ``*args`` and ``**kwargs``
+as the last parameters (this is required by the internal behaviour of the ``setup_transformations`` function).
+As previously, name of the function should match name of the filled parameter, but we can specify parameter name
+and returned type in the decorator:
+
+.. code-block:: python
+
+    @observation(observation_name='pc', observation_type=gym.spaces.Box(0.0, 1.0, (1,)))
+    def collision_probability(self, n_wifi: int, *args, **kwargs) -> float:
+        return 0.154887 * np.log(1.03102 * n_wifi)
+
+Full source code of the IEEE 802.11ax extension can be found :ref:`here <https://github.com/m-wojnar/reinforced-lib/blob/main/reinforced_lib/exts/ieee_802_11_ax.py>`.
```

Comment:
link nie działa. Znalazłem że linki do zewnętrznych stron mają taki syntax:  
\`link \<https://somecoolwebsite>`

(Tutaj więcej: https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages)